### PR TITLE
buildkit 0.6.4

### PR DESCRIPTION
**Pull request by bayandin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Bump version + Remove GOPATH (https://github.com/Homebrew/homebrew-core/issues/47627)

### CODE REVIEW

**Review by chenrui333 — APPROVED** [review]

**Comment by bayandin** [comment]
Fails on CI:
```
==> go build -ldflags -s -w -X github.com/moby/buildkit/version.Version=0.6.4 -X github.com/moby/buildkit/version.Revision=ebcef1f69af0bbca077efa9a960a481e579a0e89 -X github.com/moby/buildkit/version.Package=github.com/moby/buildkit -trimpath -o /usr/local/Cellar/buildkit/0.6.4/bin/buildctl
go: github.com/containerd/containerd@v1.3.0-0.20190507210959-7c1e88399ec0: invalid pseudo-version: version before v1.3.0 would have negative patch number
```

File: Formula/buildkit.rb [hunk 1]
```diff
@@ -14,14 +15,18 @@ class Buildkit < Formula
...
+      -X github.com/moby/buildkit/version.Revision=#{revision}
+      -X github.com/moby/buildkit/version.Package=github.com/moby/buildkit
+    ]
+    system "go", "build", "-ldflags", ldflags.join(" "), "-trimpath", "-o", bin/"buildctl"
 
-    ldflags = ["-X github.com/moby/buildkit/version.Version=#{version}",
-               "-X github.com/moby/buildkit/version.Package=github.com/moby/buildkit"]
-    system "go", "build", "-o", bin/"buildctl", "-ldflags", ldflags.join(" "), "github.com/moby/buildkit/cmd/buildctl"
+    doc.install Dir["docs/*.md"]
+    prefix.install_metafiles
```
Comment (dawidd6) [inline Formula/buildkit.rb]: I believe this line is not needed.

File: Formula/buildkit.rb [hunk 2]
```diff
@@ -14,14 +15,18 @@ class Buildkit < Formula
...
+    revision = Utils.popen_read("git rev-parse HEAD").chomp
 
-    (buildpath/"src/github.com/moby/buildkit/").install Dir["*"]
+    ldflags = %W[
+      -s -w
+      -X github.com/moby/buildkit/version.Version=#{version}
+      -X github.com/moby/buildkit/version.Revision=#{revision}
+      -X github.com/moby/buildkit/version.Package=github.com/moby/buildkit
+    ]
+    system "go", "build", "-ldflags", ldflags.join(" "), "-trimpath", "-o", bin/"buildctl"
```
Comment (dawidd6) [inline Formula/buildkit.rb]: Add `-mod=vendor`, as upstream seems to make use of dependencies vendoring.

Comment (bayandin) [reply Formula/buildkit.rb]: With `-mod=vendor` the command produces a non-executable file which even not a real binary.

Comment (bayandin) [reply Formula/buildkit.rb]: That was my stupid mistake, fixed now! :)

**Pull request closed without merging**